File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}

        for index, num in enumerate(nums):
            delta = target - num
            if delta in hash:
                return [index, hash[delta]]
            
            hash[num] = index
        
        return [-1, -1]

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    numberHash := make(map[int]int)

    for index, value := range nums {
        numberHash[value] = index
    }

    for index, value := range nums {
        mapEntry, exists := numberHash[target-value]

        if exists && mapEntry != index {
            return []int{index, mapEntry}
        }
    }

    return []int{0, 0}
}